cpp/src/io/parquet/page_data.cu

```diff
@@ -277,6 +279,7 @@ CUDF_KERNEL void __launch_bounds__(decode_block_size)
     }
     // this needs to be here to prevent warp 3 modifying src_pos before all threads have read it
     __syncthreads();
+    auto const tile32 = cg::tiled_partition<cudf::detail::warp_size>(cg::this_thread_block());
     if (t < 32) {
```

Comment:
would be nice to use `tile32` here since we already have it, but I'm not convinced it can be done in a simple way.

Reply:
We could replace this `t < 32` with `tile_warp.meta_group_rank() == 0` and it should be good but the logic at L289 would be messy since it may be tile_warp.meta_group_rank() == 0 or 1` depending on `out_threads0 == 32 or 64`.

Reply:
We could/should probably port the whole logic to `thread_group`s and avoid the magic 32 multiples. I'd expect that it would not be more complex than the current logic.
Not something for this PR.